src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx

```diff
@@ -28,12 +31,12 @@ function VideoPopoverMenuContextProvider({children}: ChildrenProps) {
     );
 
     const downloadAttachment = useCallback(() => {
-        if (currentlyPlayingURL === null) {
+        if (videoPopoverMenuPlayerRef.current === null) {
             return;
         }
-        const sourceURI = addEncryptedAuthTokenToURL(currentlyPlayingURL);
+        const sourceURI = addEncryptedAuthTokenToURL((videoPopoverMenuPlayerRef.current.props.source as AVPlaybackSourceObject).uri);
```

Comment:
I'd prefer if we didn't typecast. Its not necessary here and defeats the purpose of using TypeScript. Can you apply the following patch to make this work without typecasting and then re-test it to make sure it still works?

cc: @rayane-djouah 

```diff
diff --git a/src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx b/src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx
index b74a26caa6..14bf24c548 100644
--- a/src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx
+++ b/src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx
@@ -1,4 +1,3 @@
-import type {AVPlaybackSourceObject} from 'expo-av';
 import React, {useCallback, useContext, useMemo, useRef, useState} from 'react';
 import * as Expensicons from '@components/Icon/Expensicons';
 import type {PopoverMenuItem} from '@components/PopoverMenu';
@@ -31,10 +30,13 @@ function VideoPopoverMenuContextProvider({children}: ChildrenProps) {
     );
 
     const downloadAttachment = useCallback(() => {
-        if (videoPopoverMenuPlayerRef.current === null) {
+        const { source } = videoPopoverMenuPlayerRef.current?.props ?? {};
+
+        if (typeof source === 'number' || !source) {
             return;
         }
-        const sourceURI = addEncryptedAuthTokenToURL((videoPopoverMenuPlayerRef.current.props.source as AVPlaybackSourceObject).uri);
+
+        const sourceURI = addEncryptedAuthTokenToURL(source.uri);
         fileDownload(sourceURI);
     }, [videoPopoverMenuPlayerRef]);
 

```